From: Charles <charles05@canonical.com>
Date: Mon, 6 Jul 2026 23:30:00 +0100
Subject: tests: don't hang when mutter teardown returns non-zero

run-rdp-tests.py is hanging when mutter shutdowns with a non-zero
return code. It doesn't log anything either. Be more lenient to try
and see why mutter isn't existing cleanly in the PPA builders.

Forwarded: no
---
 tests/run-rdp-tests.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/tests/run-rdp-tests.py b/tests/run-rdp-tests.py
index f7cedbe..d594976 100755
--- a/tests/run-rdp-tests.py
+++ b/tests/run-rdp-tests.py
@@ -83,9 +83,10 @@ def stop_mutter():
   print("Stopping mutter")
   mutter.terminate()
   print("Waiting for mutter to terminate")
-  if mutter.wait() != 0:
-    print("Mutter exited incorrectly")
-    sys.exit(1)
+  mutter.wait()
+  if mutter.returncode != 0:
+    print("Mutter teardown returned %d (ignored)" % (mutter.returncode,),
+          flush=True)
   print("Done")
   loop.quit()
 
